### PR TITLE
feat: support to show inputs dir artifacts in UI

### PR DESCRIPTION
**Pull request by shuangkun**
I want to set inputs artifacts which is dir type in UI.
The example is:
```
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: artifactory-artifact-
spec:
  entrypoint: artifact-example
  templates:
  - name: artifact-example
    steps:
    - - name: generate-artifact
        template: whalesay
    - - name: consume-artifact
        template: print-message
        arguments:
          artifacts:
          - name: message
            from: "{{steps.generate-artifact.outputs.artifacts.hello-art}}"
          - name: log
            from: "{{steps.generate-artifact.outputs.artifacts.log}}"

  - name: whalesay
    inputs:
      artifacts:
      - archive:
          none: {}
        name: danger
        oss:
          key: aaaaa/store/artifactory-artifact-647sm/log
        path: /tmp
    container:
      image: docker/whalesay:latest
      command: [sh, -c]
      args: ["cowsay hello world | tee /tmp/hello_world.txt && cowsay hello world | tee /tmp/hello_world2.txt"]
    outputs:
      artifacts:
      - name: hello-art
        path: /tmp
      - archive:
          none: {}
        name: log
        oss:
          key: aaaaa/store/{{workflow.name}}/log
        path: /tmp

  - name: print-message
    inputs:
      artifacts:
      - name: message
        path: /tmp/message
      - name: log
        path: /tmp/inputs/artifacts/logs
    container:
      image: alpine:latest
      command: [sh, -c]
      args: ["ls /tmp/message && ls /tmp/inputs/artifacts/logs"]
```
The UI is when I click inputs artifacts which is a dir:
image

image

Fixes #TODO

### CODE REVIEW

File: server/artifacts/artifact_server.go [hunk 1]
```diff
@@ -95,7 +95,7 @@ func (a *ArtifactServer) GetArtifactFile(w http.ResponseWriter, r *http.Request)
 	direction := requestPath[directionIndex]
 	artifactName := requestPath[artifactNameIndex]
 
-	if direction != "outputs" { // for now we just handle output artifacts
+	if direction != "outputs" && direction != "inputs" { // for now we handle output and input artifacts
```
Comment (isubasinghe) [inline server/artifacts/artifact_server.go:98]: Update the comments above the function.

Comment (isubasinghe) [reply server/artifacts/artifact_server.go:98]: Probably a good idea to introduce a string enum at this stage, just makes a it a bit more structured

```go
type Direction string

const (
	Outputs Direction = "outputs"
	Inputs Direction = "inputs"
)
```

**Review by isubasinghe — CHANGES_REQUESTED** [review]
Thanks for the feature, code looks good. I've requested some slight changes.

File: server/artifacts/artifact_server.go (hunk 1 above)
Comment (shuangkun) [reply server/artifacts/artifact_server.go]: okay, I have change it.

**Review by isubasinghe — CHANGES_REQUESTED** [review]
Sorry maybe I should've been more clear: 
These comments need to be updated: https://github.com/argoproj/argo-workflows/blob/6adf0a077d1a36b9c487f336a1d2d075b8eef9c0/server/artifacts/artifact_server.go#L69

**Comment by shuangkun** [comment]
> Sorry maybe I should've been more clear: These comments need to be updated:
> 
> https://github.com/argoproj/argo-workflows/blob/6adf0a077d1a36b9c487f336a1d2d075b8eef9c0/server/artifacts/artifact_server.go#L69

Okay, I've added it.

File: server/artifacts/artifact_server.go [hunk 2]
```diff
@@ -62,6 +69,9 @@ func (a *ArtifactServer) GetInputArtifact(w http.ResponseWriter, r *http.Request
 //	/artifact-files/{namespace}/[archived-workflows|workflows]/{id}/{nodeId}/outputs/{artifactName}
 //	/artifact-files/{namespace}/[archived-workflows|workflows]/{id}/{nodeId}/outputs/{artifactName}/{fileName}
 //	/artifact-files/{namespace}/[archived-workflows|workflows]/{id}/{nodeId}/outputs/{artifactName}/{fileDir}/.../{fileName}
+//	/artifact-files/{namespace}/[archived-workflows|workflows]/{id}/{nodeId}/inputs/{artifactName}
+//	/artifact-files/{namespace}/[archived-workflows|workflows]/{id}/{nodeId}/inputs/{artifactName}/{fileName}
+//	/artifact-files/{namespace}/[archived-workflows|workflows]/{id}/{nodeId}/inputs/{artifactName}/{fileDir}/.../{fileName}
```
Comment (isubasinghe) [inline server/artifacts/artifact_server.go:74]: Probably best to stick to the existing format?
[inputs|outputs]

like [archived-workflows|workflows]

Comment (shuangkun) [reply server/artifacts/artifact_server.go]: Okay, sounds better.

**Review by isubasinghe — APPROVED** [review]
LGTM

**Review by terrytangyuan — APPROVED** [review]

**Comment by terrytangyuan** [comment]
Thanks. LGTM

**Pull request merged**